### PR TITLE
2x: Fix wording in Async and Publish processors javadoc.

### PR DESCRIPTION
**Pull request by serj-lotutovici**
Seams that these are some leftovers from the 1.x -> 2.x refactoring.

### CODE REVIEW

File: src/main/java/io/reactivex/processors/PublishProcessor.java [hunk 1]
```diff
@@ -23,14 +23,14 @@
 import io.reactivex.plugins.RxJavaPlugins;
 
 /**
- * A Subject that multicasts events to Subscribers that are currently subscribed to it.
+ * Processor that multicasts all subsequently observed items to its current {@link Subscriber}s.
```
Comment (akarnokd) [inline src/main/java/io/reactivex/processors/PublishProcessor.java:26]: "A Processor"  ?

**Review by akarnokd — APPROVED** [review]

File: src/main/java/io/reactivex/processors/PublishProcessor.java (hunk 1 above)
Comment (serj-lotutovici) [reply src/main/java/io/reactivex/processors/PublishProcessor.java:26]: In Replay and Behaviour processors it's "Processor" \_(ツ)_/¯. Both seam syntactically correct.

Comment (akarnokd) [reply src/main/java/io/reactivex/processors/PublishProcessor.java:26]: Then the `AsyncProcessor` could be adjusted as well.

**Review by vanniktech — APPROVED** [review]

File: src/main/java/io/reactivex/processors/PublishProcessor.java (hunk 1 above)
Comment (serj-lotutovici) [reply src/main/java/io/reactivex/processors/PublishProcessor.java:26]: done

**Pull request merged**